src/storage/local_table.cpp

```diff
@@ -190,11 +192,56 @@ void StringLocalColumn::commitLocalChunkOutOfPlace(
     column->append(stringColumnChunk, startPageIdx, nodeGroupIdx);
 }
 
+void VarListLocalColumn::prepareCommitForChunk(node_group_idx_t nodeGroupIdx) {
+    auto chunk = chunks.at(nodeGroupIdx).get();
+    auto varListColumn = reinterpret_cast<VarListNodeColumn*>(column);
+    auto listColumnChunkInStorage = ColumnChunkFactory::createColumnChunk(column->getDataType());
+    auto columnChunkToUpdate = ColumnChunkFactory::createColumnChunk(column->getDataType());
+    varListColumn->scan(nodeGroupIdx, listColumnChunkInStorage.get());
+    common::offset_t nextOffsetToWrite = 0;
+    auto numNodesInGroup =
+        column->metadataDA->get(nodeGroupIdx, TransactionType::READ_ONLY).numValues;
+    for (auto& [vectorIdx, localVector] : chunk->vectors) {
+        auto startOffsetInChunk = StorageUtils::getStartOffsetInChunk(vectorIdx);
+        auto listVector = localVector->vector.get();
+        auto listVectorSize = listVector->state->selVector->selectedSize;
+        listVector->state->selVector->selectedSize = 1;
+        for (auto i = 0u; i < listVectorSize; i++) {
```

Comment:
This is assuming that selected positions are sorted, which is not guaranteed. We should loop over validityMask instead.